agent/api/task/service_connect_validation.go

```diff
@@ -0,0 +1,293 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: Could we call this `service_connect_validator`? I would also suggest creating an interface for Attachment, Parser and Validator and use that for all attachment types.

Reply:
Rename files to validator in the commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on the agent/api/task/service_connect_validator.go and agent/api/task/service_connect_validator_test.go. Thanks